src/lib/pack-strategy.test.ts

```
@@ -0,0 +1,221 @@
+
+describe("splitContentByTokenBudget", () => {
+  it("should return original content when budget is non-positive", () => {
+    expect(splitContentByTokenBudget("hello", 0)).toEqual(["hello"]);
+  });
+
+  it("should return original content when no split is needed", () => {
+    expect(splitContentByTokenBudget("hello", 10)).toEqual(["hello"]);
+  });
+
+  it("should split at paragraph boundaries when possible", () => {
+    const content = "a".repeat(60) + "\n\n" + "b".repeat(60);
```

Biome (2.4.4)

[error] Template literals are preferred over string concatenation.

(lint/style/useTemplate)